web/workflow.R

```diff
@@ -11,184 +11,217 @@
-if ("sensitivity.analysis" %in% names(settings)
-    && PEcAn.utils::status.check("SENSITIVITY") == 0) {
-  PEcAn.utils::status.start("SENSITIVITY")
-  runModule.run.sensitivity.analysis(settings)
-  PEcAn.utils::status.end()
-}
-
-# Run parameter data assimilation
-if ("assim.batch" %in% names(settings)) {
-  if (PEcAn.utils::status.check("PDA") == 0) {
-    PEcAn.utils::status.start("PDA")
-    settings <- PEcAn.assim.batch::runModule.assim.batch(settings)
-    PEcAn.utils::status.end()
-  }
-}
-
-# Run state data assimilation
-if ("state.data.assimilation" %in% names(settings)) {
-  if (PEcAn.utils::status.check("SDA") == 0) {
-    PEcAn.utils::status.start("SDA")
-    settings <- sda.enfk(settings)
-    PEcAn.utils::status.end()
-  }
-}
-
-# Run benchmarking
-if ("benchmarking" %in% names(settings)
-    && "benchmark" %in% names(settings$benchmarking)) {
-  PEcAn.utils::status.start("BENCHMARKING")
-  results <- papply(settings, function(x) calc_benchmark(x, bety))
-  PEcAn.utils::status.end()
-}
-
-# Pecan workflow complete
-if (PEcAn.utils::status.check("FINISHED") == 0) {
-  PEcAn.utils::status.start("FINISHED")
-  PEcAn.remote::kill.tunnel(settings)
-  db.query(paste("UPDATE workflows SET finished_at=NOW() WHERE id=",
-                 settings$workflow$id, "AND finished_at IS NULL"),
-           params = settings$database$bety)
-
-  # Send email if configured
-  if (!is.null(settings$email)
-      && !is.null(settings$email$to)
-      && (settings$email$to != "")) {
-    sendmail(settings$email$from, settings$email$to,
-             paste0("Workflow has finished executing at ", base::date()),
-             paste0("You can find the results on ", settings$email$url))
-  }
-  PEcAn.utils::status.end()
-}
-
-db.print.connections()
-print("---------- PEcAn Workflow Complete ----------")
+suppressMessages(library("PEcAn.all"))
+suppressMessages(library("PEcAn.utils"))
+suppressMessages(library("RCurl"))
+suppressMessages(library("optparse"))
+
+# --------------------------------------------------
+get_args <- function () {
+    option_list = list(
+      make_option(
+        c("-s", "--settings"),
+        default = ifelse(Sys.getenv("PECAN_SETTINGS") != "", 
+                         Sys.getenv("PECAN_SETTINGS"), "pecan.xml"),
+        type = "character",
+        help = "Settings XML file",
+        metavar = "FILE",
+      ),
+      make_option(
+        c("-c", "--continue"),
+        default = FALSE,
+        action = "store_true",
+        type = "logical",
+        help = "Continue processing",
+      )
+    )
+
+    parser = OptionParser(option_list = option_list)
+    args = parse_args(parser)
+
+    if (!file.exists(args$settings)) {
+        print_help(parser)
+        stop(sprintf('--settings "%s" not a valid file\n', args$settings))
+    }
+
+    return(invisible(args))
+}
+
+# --------------------------------------------------
+main <- function () {
+    # get command-line arguments
```

Comment:
I assume rest is just whitespaces from everything shifted?

Reply:
Yes.